v0/destinations/braze/transform.js

```diff
@@ -349,6 +349,64 @@ function processGroup(message, destination) {
       })
       .build();
   }
+  if (destination.Config.enableSubscriptionGroupInGroupCall) {
+    if (!message.traits) {
+      throw new ErrorBuilder()
+        .setStatus(400)
+        .setMessage(
+          "Message should have traits with subscriptionState, email or phone"
+        )
+        .setStatTags({
+          destType: DESTINATION,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+          stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+          meta:
+            TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+        })
+        .build();
+    }
+    const subscriptionGroup = {};
+    subscriptionGroup.subscription_group_id = groupId;
+    if (
+      message.traits.subscriptionState !== "subscribed" &&
+      message.traits.subscriptionState !== "unsubscribed"
+    ) {
+      throw new ErrorBuilder()
+        .setStatus(400)
+        .setMessage(
+          "You must provide a subscription state that is either subscribed or unsubscribed. Aborting message."
+        )
+        .setStatTags({
+          destType: DESTINATION,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+          stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+          meta:
+            TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+        })
+        .build();
+    }
+    subscriptionGroup.subscription_state = message.traits.subscriptionState;
+    subscriptionGroup.external_id = [message.userId || message.anonymousId];
+    if (getFieldValueFromMessage(message, "phone")) {
+      subscriptionGroup.phone = [getFieldValueFromMessage(message, "phone")];
+    } else if (getFieldValueFromMessage(message, "email")) {
+      subscriptionGroup.email = [getFieldValueFromMessage(message, "email")];
+    }
+    const response = defaultRequestConfig();
+    response.endpoint = getSubscriptionGroupEndPoint(
+      destination.Config.endPoint
+    );
+    response.body.JSON = removeUndefinedValues(subscriptionGroup);
+    return {
+      ...response,
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        Authorization: `Bearer ${destination.Config.restApiKey}`
+      }
+    };
```

Comment:
We should use `buildResponse` function here.

Reply:
for this use case case, this is good I believe as we are fetching diff endpoint and format is diff in buildReponse method